### PR TITLE
Add type_hints

### PR DESCRIPTION
**Pull request by wakamezake**
Add type_hints to class methods and function arguments.
I'm going to add type_hints for the return value as well.

TrackedObject's initial_detection type is undefined because it is defined in the same file as the one you want to define as a Detection, but you can't refer to it.
https://github.com/tryolabs/norfair/blob/35af6e25bb1b82d8895d8b10e98cb518236f8c33/norfair/tracker.py#L165-L177

### CODE REVIEW

**Comment by joaqo** [comment]
Thanks for the PR @wakamezake ! We were actually looking into adding types ourselves, so this PR came at just the perfect time. I'll review it as soon as I can.

File: norfair/drawing.py
Comment (joaqo) [inline norfair/drawing.py:14]: This could be a bit more specific by specifying that the Tuple should contain 3 ints.

File: norfair/drawing.py
Comment (joaqo) [inline norfair/drawing.py:11]: Instead of `List` we could be less specific here and use `Sequence `. Also adding `"Detection"` to avoid having trouble with `Detection` being defined in this same file.

File: norfair/drawing.py
Comment (joaqo) [inline norfair/drawing.py:42]: Same as with detections above, use `Sequence ` and `"TrackedObject"`.

File: norfair/drawing.py
Comment (joaqo) [inline norfair/drawing.py:94]: Same as with detections above, use `Sequence` and `"TrackedObject"`.

File: norfair/tracker.py
Comment (joaqo) [inline norfair/tracker.py:13]: Add the types which conform the distance function's arguments and output.

File: norfair/tracker.py
Comment (joaqo) [inline norfair/tracker.py:14]: The `Union` is not necessary, `float` includes `int`s as described [here](https://twitter.com/llanga/status/1306391094847057920). Thanks @erovira for the tip!

File: norfair/tracker.py
Comment (joaqo) [inline norfair/tracker.py:18]: The Union is not necessary, float includes ints as described here.

File: norfair/tracker.py
Comment (joaqo) [inline norfair/tracker.py:29]: Should use `Sequence` instead of `List` and `"Detection"`.

File: norfair/tracker.py
Comment (joaqo) [inline norfair/tracker.py:64]: Same as mentioned above.

File: norfair/tracker.py
Comment (joaqo) [inline norfair/tracker.py:176]: Same as mentioned previously.

File: norfair/utils.py
Comment (joaqo) [inline norfair/utils.py:23]: Same as mentioned previously.

File: norfair/utils.py
Comment (joaqo) [inline norfair/utils.py:44]: Could specify that they are `int`s.

**Review by joaqo — CHANGES_REQUESTED** [review]
I had some time to check the PR. I left some comments in your code.

Also, running the following commands in Norfair's root folder:

```bash
> pip install mypy
> mypy .  --ignore-missing-imports
```

returns the following typing issues which should hopefully be fixed before merging this PR:

```bash
norfair/tracker.py:20: error: Need type annotation for 'tracked_objects' (hint: "tracked_objects: List[<type>] = ...")
norfair/tracker.py:185: error: Incompatible types in assignment (expression has type "float", variable has type "int")
norfair/drawing.py:64: error: Incompatible types in assignment (expression has type "Tuple[Any, ...]", variable has type "Tuple[int, int, int]")
norfair/drawing.py:65: error: Incompatible types in assignment (expression has type "Tuple[Any, ...]", variable has type "Tuple[int, int, int]")
norfair/drawing.py:127: error: Incompatible types in assignment (expression has type "Tuple[Any, ...]", variable has type "Tuple[int, int, int]")
norfair/video.py:85: error: Argument 1 to "Progress" has incompatible type "*List[object]"; expected "Union[str, ProgressColumn]"
norfair/video.py:144: error: "None" has no attribute "write"
norfair/video.py:173: error: Missing return statement
Found 8 errors in 3 files (checked 7 source files)
```

Finally, regarding your problem with using the type `Detection` before it is defined, you can use the string `"Detection"` in its place , as described [here](https://realpython.com/python-type-checking/#classes-as-types).

Thanks a lot for the PR. It should be ready to merge after these changes.

File: norfair/utils.py
Comment (wakamezake) [reply norfair/utils.py:23]: Calling TrackedObject in utils.py causes import error.
You can see the error in the github actions at the following link.
https://github.com/tryolabs/norfair/runs/1152822142?check_suite_focus=true

**Comment by joaqo** [comment]
Merged, thanks for the PR!

File: norfair/utils.py
Comment (joaqo) [reply norfair/utils.py]: Yeah we gotta add `from .tracker import TrackedObjec` to utils.py. Dont worry i'll fix it.

Comment (joaqo) [reply norfair/utils.py]: Turns out its harder than I thought as its triggering circular imports.

**Comment by joaqo** [comment]
Making mypy run with no errors turned to be a bit harder than expected, but now its running. Commits https://github.com/tryolabs/norfair/commit/1753adbf66b8431054dc75e676bd9bf9c62129f2 and https://github.com/tryolabs/norfair/commit/c8c8b84e36ffc0e0f37e5a7f188915c8ffb0ff71 for anyone interested.

**Pull request merged**
